src/Workspaces/Core/Portable/CodeFixesAndRefactorings/IFixAllState.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Internal.Log;
+using FixAllScope = Microsoft.CodeAnalysis.CodeFixes.FixAllScope;
+
+namespace Microsoft.CodeAnalysis.CodeFixesAndRefactorings
+{
+    internal interface IFixAllState
+    {
+        int CorrelationId { get; }
+        IFixAllProvider FixAllProvider { get; }
+        string? CodeActionEquivalenceKey { get; }
+        object Provider { get; }
```

Comment:
doc this guy.

Reply:
1. nit: if would be nice if the original providers might be able to have a shared interface as well.
2. consider naming this more distinctly to prevent confusion with FixAllProvider.

Reply:
Added doc.

> nit: if would be nice if the original providers might be able to have a shared interface as well.

I think this would be a good follow-up work item as there are places outside of this property where we pass along `object` instead of a strong type for providers.

> consider naming this more distinctly to prevent confusion with FixAllProvider.

I thought about this. Calling it `FixProvider` seemed to confuse with `FixAllProvider` and `UnderlyingProvider` or `UnderlyingFixProvider` seemed unnecessarily verbose. So, left it as `Provider`, and now have a doc comment. Let me know if you have a preferred name.